src/main/java/com/sele3/page/vietJet/SelectFlightPage.java

```diff
@@ -0,0 +1,197 @@
+package com.sele3.page.vietJet;
+
+import com.codeborne.selenide.Condition;
+import com.codeborne.selenide.ElementsCollection;
+import com.codeborne.selenide.SelenideElement;
+import com.sele3.data.vietjet.Passenger;
+import com.sele3.data.vietjet.TicketData;
+import com.sele3.enums.vietjet.LabelType;
+import com.sele3.utils.Constants;
+import com.sele3.utils.YamlUtils;
+import io.qameta.allure.Step;
+
+
+import java.time.LocalDate;
+import java.time.MonthDay;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.codeborne.selenide.Selenide.*;
+
+public class SelectFlightPage {
+
+    public boolean allPricesHaveVND() {
+        ElementsCollection prices = $$("p.MuiTypography-body1");
+
+        for (SelenideElement price : prices) {
+            if (!price.getText().contains("VND")) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean isSelectTravelOptionsPageDisplayed() {
+        SelenideElement flights = $$("div")
+                .findBy(Condition.text("VJ"));
+        return flights.isDisplayed();
+    }
+
+    public int getCheapestPrice() {
+        return price.stream()
+                .map(el -> {
+                    String full = el.scrollIntoView(true).getText();
+                    return full.replaceAll("[^\\d]", "");
+                })
+                .mapToInt(Integer::parseInt)
+                .min()
+                .orElseThrow(() -> new RuntimeException("No price found"));
+    }
+
+    public void scrollUntilAllPricesLoaded() {
+        int unchangedCount = 0;
+        int previousSize = 0;
+
+        while (unchangedCount < 3) {
+            price.last().scrollIntoView(true);
+            sleep(500);
+
+            int currentSize = price.size();
+            if (currentSize == previousSize) {
+                unchangedCount++;
+            } else {
+                unchangedCount = 0;
+                previousSize = currentSize;
+            }
+        }
+    }
+
+    @Step("Select the first cheapest ticket")
+    public void selectTheFirstCheapestTicket() {
+        scrollUntilAllPricesLoaded();
+        int minPrice = getCheapestPrice();
+        String formattedPrice = String.format("%,d", minPrice / 1000);
+        $$x(String.format("//p[contains(text(),'%s')]", formattedPrice)).get(0)
+                .scrollIntoView(false)
+                .shouldBe(Condition.visible)
+                .click();
+        clickContinueButton();
+    }
+
+    @Step("Click on Continue button")
+    public void clickContinueButton() {
+        getButtonByText((String) YamlUtils.getProperty("button.continue")).click();
+    }
+
+    public String getFromPlace() {
+        return $x(String.format(textByLabel, LabelType.FROM)).getText().split("\\(")[0].trim();
```

Comment:
textByLabel element's locator is dynamic too much in this case. Please write a better precise locator